perception_node: # Name of Node
  ros__parameters:
    # left_camera_topic: "/left_camera_node/images/compressed"
    # right_camera_topic: "/right_camera_node/images/compressed"
    left_camera_topic: "synced_left_image"
    right_camera_topic: "synced_right_image"
    cone_detections_topic: "perception/cone_detections"
    processed_lidar_topic: "/lidar_pipeline/detected"
    baseline: 10.0
    update_rate: 33.0 #[ms]
    debug: True
    distortion: [-0.269633921151086, 0.12944192627838266, 0.0, 0.0, 0.0]
    intrinsics:
      [
        1169.929054195447,
        0.0,
        732.7234047220617,
        0.0,
        1168.6955521426203,
        539.5227062057302,
        0.0,
        0.0,
        1.0,
      ]
    rotation:
      [
        0.9948,
        -0.0030,
        -0.1020,
        0.0040,
        0.9999,
        0.0094,
        0.1020,
        -0.0097,
        0.9947,
      ]
    translation: [-516.6818, 29.1172, -386.7437]
    rectify:
      [
        0.999933395,
        0.000876606484,
        0.0115081202,
        -0.000858846762,
        0.999998433,
        -0.00154808535,
        -0.0115094592,
        0.00153809852,
        0.999932581,
      ]
    save_pic: "False"
    confidence: 0.40
    cone_heights: [0.0, 325.0, 325.0, 325.0, 505.0]
    lidar_only_detection: False

lidar_proc_node:
  ros__parameters:
    update_rate: 100.0 #[ms]
    debug: False
    topics: # TODO: fix
      input: "/ouster/points"
      filtered: "/lidar_pipeline/filtered"
      no_ground: "/lidar_pipeline/no_ground"
      clustered: "/lidar_pipeline/clustered"
      detected: "/lidar_pipeline/detected"
      heartbeat: "/lidar_pipeline/heartbeat"
    view_filter:
      outer_box: #remove all points outside of outer box
        xmin: 0.0
        xmax: 30.0
        ymin: -12.0
        ymax: 12.0
      inner_box: #remove all points within inner box
        xmin: -1.013
        xmax: 2.1
        ymin: -0.8
        ymax: 0.8
      grid_size_x: 40 #create grid of lowest z points for each cell in grid
      grid_size_y: 20
    ground_ransac: #use z points for ransac
      max_iterations: 300
      distance_threshold: 0.15
    clustering:
      cluster_tolerance: 0.35
      min_cluster_size: 2 #3
      max_cluster_size: 50
      reconstruct_radius: 0.25
    cone_filter:
      cone_height: 0.387
      cone_radius: 0.075
      big_cone_height: 0.6216
      big_cone_radius: 0.11
      mse_threshold: 0.00003 #any point as close as threshold is inlier
      lin_threshold: 0.022
      IOUThreshold: 0.06
